Applications/Spire/Source/Ui/TextBox.cpp

```diff
@@ -157,83 +187,123 @@ void TextBox::keyPressEvent(QKeyEvent* event) {
 
 void TextBox::paintEvent(QPaintEvent* event) {
   auto computed_style = compute_style();
-  auto style = QString("QLineEdit {");
-  style += "border-style: solid;";
+  auto placeholder_style = QString(
+    R"(QLabel {
+      background: transparent;
+      border-color: transparent;)");
+  auto line_edit_style = QString("QLineEdit {");
+  line_edit_style += "border-style: solid;";
   for(auto& property : computed_style.get_properties()) {
     property.visit(
       [&] (const BackgroundColor& color) {
-        style += "background-color: " +
+        line_edit_style += "background-color: " +
           color.get_expression().as<QColor>().name(QColor::HexArgb) + ";";
       },
       [&] (const BorderTopSize& size) {
-        style += "border-top-width: " + QString::number(
+        auto property = "border-top-width: " + QString::number(
```

Comment:
Because the QLabel fills the whole TextBox, it is impossible to have different size, borders and background colors between placeholders and TextBox like CSS. You can test different placeholder styles in CSS from the link below:
https://www.w3schools.com/cssref/tryit.asp?filename=trycss3_placeholder

Since our Textbox doesn’t have this spec, maybe I am overthinking.

Reply:
Fixed by making the placeholder a pseudo-element.